src/Illuminate/Validation/Validator.php

```diff
@@ -619,8 +619,14 @@ protected function presentOrRuleIsImplicit($rule, $attribute, $value)
      */
     protected function passesOptionalCheck($attribute)
     {
+        $data = Arr::dot($this->initializeAttributeOnData($attribute));
+
+        $data = array_merge($data, $this->extractValuesForWildcards(
+            $data, $attribute
+        ));
+
```

Comment:
@themsaid  `$data` is needed only when the attribute has `sometimes` rule. Could you move data extraction logic inside if condition?